src/test/java/org/javarosa/plugins/InstancePluginTest.java

```diff
@@ -86,21 +86,36 @@ public void instanceProvider_supportsPartialElements() throws IOException, XForm
         assertThat(firstItem.isPartial(), equalTo(true));
         assertThat(firstItem.getNumChildren(), equalTo(2));
         assertThat(firstItem.getChildAt(0).getName(), equalTo("value"));
-        assertThat(firstItem.getChildAt(0).getValue(), equalTo(null));
+        assertThat(firstItem.getChildAt(0).getValue(), equalTo(new StringData("0")));
```

Comment:
I think I'd prefer to keep the existing tests as simpler and just test that we support partial elements and then have the last test use a different fake (or make the existing one configurable) to set up the special case where the first element in a nodeset is partial.